csrc/scheduler/matmul_utils.cpp

```diff
@@ -202,6 +352,29 @@ inline bool initExtraHeuristics(
   return true;
 }
 
+MatmulLayout inferLayout(Fusion* fusion, const MmaOp* mma_expr) {
+  const auto& mma_inputs = mma_expr->inputs();
+  TORCH_INTERNAL_ASSERT(mma_expr->layout(), "Failed to acquire inputs layout.");
+  const MatmulLayout matmul_layout = *mma_expr->layout();
+  auto [transA, transB] = MmaLayoutToBool(matmul_layout);
+  auto loadsA =
+      getAllFilteredExprsBetween<LoadStoreOp>(fusion->inputs(), mma_inputs[0]);
```

Comment:
@zasdfgbnm what would be the best way to get the layout in the scheduler after you merged https://github.com/NVIDIA/Fuser/pull/234? I'm counting transposing `LoadStoreOp`s, on top of the mma layout, but doesn't look perfect.

Reply:
A better way is to use a CA map, something like
```c++
ComputeAtMap ca_map(fusion);
ca_map.areMapped(mma_input_last_id, fusion_input_last_id, IdMappingMode::PERMISSIVE);
```

The CA map is a powerful tool that contains information about the equivalence relationship of IterDomains of each dimension of each tensor. For example, if you have
```c++
T1[I4, I5, I6] = transpose(T0[I1, I2, I3], 1, 2);
T2[I7, I8, I9] = set(T1[I4, I5, I6]);
```
then the CA map will be
```
DisjointSets {
  {I7, I4, I1},
  {I8, I5, I3},
  {I9, I6, I2}
}
```
Two IDs in the same disjoint set are called they are "mapped".

There are multiple different mapping mode, each has different details. `PERMISSIVE` here means allow broadcast concretization. For example, in the permissive map,
```
T2[I5, I6] = T1[I1, I2] + T2[I3, b4]
```
`I6`, `I2`, `b4` are mapped to each other, but this is not the case for "exact map".